dissect/etl/headers/headers.py

```diff
@@ -204,7 +210,7 @@ def minimal_size(self) -> int:
     def _header_type(self) -> Structure:
```

Comment:
```suggestion
    def _header_type(self) -> c_etl_headers.MessageHeader:
```

Reply:
The other `Structure` types below can also be made more specific.

Given `ErrorHeader._header_type()` returns `None`, the base class `Header._header_type()` return type should probably be changeds to `Structure | None`.